contributors/design-proposals/storage/online-grow-volume-size.md

```diff
@@ -0,0 +1,252 @@
+	- [Online File System Resize in Kbuelet](#online-file-system-resize-in-kbuelet)
+		- [Volume Resize Request Detect](#volume-resize-request-detect)
+		- [File System Resizing Operation](#file-system-resizing-operation)
+
+## Overview
+
+Release 1.10 only supports offline file system resizing for PVCs, as this operation is only executed inside the `MountVolume` operation in kubelet. If a resizing request was submitted after the volume mounted, it won't be performed. This proposal intent to support online file system resizing for PVCs in kubelet.
+
+## Goals
+
+Enable users to increase size of PVC which already in use(mounted). The user will update PVC for requesting a new size.  Underneath we expect that - according kubelet will resize the file system for the PVC.
+
+## Non Goals
+
+* Offline file system resizing not included. If we found a volume need file system resizing but not mounted to the node yet, we will do nothing. This situation will be dealt with by the existing [offline file system resizing handler](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/grow-volume-size.md).
+
+* Extending resize tools: We only support for most common file systems' offline resizing in current release, and we prefer to stay the same for the online one.
+
+## Use Cases
+
+* As a user I am running Mysql on a 100GB volume - but I am running out of space, I should be able to increase size of volume mysql is using without losing all my data. (online and with data)
+* As a user I am running an application with a PVC. I should be able to resize the volume without losing data or mount point. (online and with data and without taking pod offline)
+
+## Prerequisite
+
+- Currently we only support offline resizing for `xfs`, `ext3`, `ext4`. Online resizing of `ext3`,`ext4` is introduced from [Linux kernel-3.3](https://www.ibm.com/developerworks/library/l-33linuxkernel/), and `xfs` always supported growing mounted partitions (in-fact currently there is no way to expand an unmounted `xfs` file system), so they are all safe for online resizing. If user tries to expand a volume with other formats an error event will be reported for according pod.
+
+- This feature will be protected by an alpha feature gate `ExpandOnlinePersistentVolumes` in v1.11. We separate this feature gate from the offline resizing gate `ExpandPersistentVolumes`, if user wants to enable this feature, `ExpandPersistentVolumes` should be enabled first.
+
+## Implementation Design
+
+### Design Overview
+
+The key point of online file system resizing is how kubelet to discover which  PVCs need file system resizing. We achieve this goal by adding a PVC cache for kubelet. In detail, when kubelet receives a new pod, it adds all PVCs mounted by this pod to the cache, and  remove according PVCs when a pod is deleted. The PVC information is cached there with a TTL to reduce the load on both `etcd` and `apiserver`.
```

Comment:
@liggitt, heads up: this proposal adds PVC cache into Kubelet, just as it has Secrets and ConfigMap caches. It will refresh (GET) all PVCs used by running or scheduled pods every now and then.

Should we worry about performance / API server load?

Reply:
Is the TTL going to be set according the cluster size similar to that for Secrets and ConfigMaps? I think in general we do have concerns about the scalability of these requests. The hope (AFAIK) is to switch to "watch" if the security concerns can be addressed. 

/cc @wojtek-t for the scalability issues.

Reply:
Isn't there a cost associated with watches as well? I have seen some production failures where number of watches pushed etcd over the edge.  If watches is the way to go, the watch has to apply selectively and should not be cluster wide - right?

Reply:
@yujuhong  Yeah, the TTL is similar to that for Secrets and ConfigMaps.